htlcswitch/link.go

```diff
@@ -1009,11 +1009,14 @@ out:
 			break out
 		}
 
-		// If the previous event resulted in a non-empty
-		// batch, reinstate the batch ticker so that it can be
-		// cleared.
+		// If the previous event resulted in a non-empty batch, resume
+		// the batch ticker so that it can be cleared. Otherwise pause
+		// the ticker to prevent waking up the htlcManager while the
+		// batch is empty.
```

Comment:
Should we be using a timer instead?

Reply:
It could be a timer, but also the ticker is currently used pretty heavily in the unit tests that cover this code. So we'd need something similar to `time.After` that also allows manual triggering. I don't really want to extend the scope of this pr to include that.